src/main/java/com/github/mavogel/ilias/lib/wrapper/soap/SoapEndpoint.java

```diff
@@ -365,4 +367,59 @@ public void setRegistrationDatesOnGroups(final List<IliasNode> groups, final Loc
             }
         }
     }
+
+    /**
+     * Adds a new group to a given course.
+     * If the group already exists, the user will be asked, if he
+     * still wants to create it. Unfortunately the exist mechanism searches in all
+     * courses of the user and not just in the given one. We cannot restrict this search, thus
+     * the user has to decide. (@see {@link com.github.mavogel.ilias.lib.wrapper.IliasEndpoint})
+     *
+     * @param course    The course the group shall be created in
+     * @param groupName The name of the group
+     * @return True iff the group has been successfully been created
+     * @throws Exception in case of a failure. Detailed logs are written.
+     */
+    @Override
+    public boolean addGroup(IliasNode course, String groupName) throws Exception {
+        if (endpoint.groupExists(userDataIds.getSid(), groupName)) {
+            LOG.info("Group \"" + groupName + "\" already exists, but unfortunately we cannot check in which course. Please check it yourself. Do you still want to create this group?");
+            if (!(readAndParseUserConfirmation())) {
+                return false;
+            }
+        }
+        try {
+            int result = endpoint.addGroup(userDataIds.getSid(), course.getRefId(), SoapXMLUtils.generateAddGroupXMLFromName(groupName));
+            LOG.info("Created group \"" + groupName + "\"");
+            LOG.debug("With ref_id: " + result);
+            return true;
+        } catch (RemoteException e) {
+            LOG.error("Could not add group " + groupName);
+            return false;
+        }
+    }
+
+    //TODO: Taken from IOUtils from the ilias-client. Maybe move the class to this lib and put this method there?
+    private static boolean readAndParseUserConfirmation() {
```

Comment:
Remove this method and let it in the `ilias-client`

Reply:
Done